OpenLaTeXStudio/EditorModule/src/latexstudio/editor/settings/ApplicationSettings.java

```diff
@@ -107,12 +117,76 @@ public boolean getLineWrapStatus(){
         return Boolean.parseBoolean(getProperty(LINEWRAP_STATUS));
     }
     
-    public void setAutoCompleteStatus(boolean autoCompleteStatus){
-        pcs.firePropertyChange(AUTOCOMPLETE_STATUS, getAutoCompleteStatus(), autoCompleteStatus);
-        setProperty(AUTOCOMPLETE_STATUS, String.valueOf(autoCompleteStatus));
+    public void registerSettingListeners(Object o) {
+        for(Method m : o.getClass().getMethods()) {
+            for(Annotation a : m.getAnnotations()) {

```

Comment:
Method exposes "getAnnotation(Class<T>)", which allows to specify the annotation we wish to retrieve. This would simplify a lot in this method.